File: CS/Form1.cs
using System;
using System.Collections;
using System.Windows.Forms;
using System.ComponentModel.Design;
using DevExpress.XtraBars;
using DevExpress.XtraReports.UserDesigner;
// ...

namespace FieldList_UpdateDataSource {
    public partial class Form1 : Form {

        public Form1() {
            InitializeComponent();
        }

        const string xmlPath = @"..\..\Cars.xml";

        private void Form1_Load(object sender, EventArgs e) {
            xrDesignPanel1.ExecCommand(ReportCommand.NewReport);
        }

        private void commandBarItem46_ItemClick(object sender, ItemClickEventArgs e) {
            BindReportToData();
        }

        #region #UpdateFieldList
        private void BindReportToData() {
            if (xrDesignPanel1.Report == null)
                return;
             // Create a data source and bind it to a report.
            xrDesignPanel1.Report.DataSource = CreateDataSource();

            // Update the Field List.
            FieldListDockPanel fieldList =
                (FieldListDockPanel)xrDesignDockManager1[DesignDockPanelType.FieldList];
            IDesignerHost host = 
                (IDesignerHost)xrDesignPanel1.GetService(typeof(IDesignerHost));
            fieldList.UpdateDataSource(host);
        }
        #endregion #UpdateFieldList

        private ArrayList CreateDataSource() {
            // Create a list.
            ArrayList list = new ArrayList();

            // Populate the list with records.
            list.Add(new Record(1, "Jane", 19));
            list.Add(new Record(2, "Joe", 30));
            list.Add(new Record(3, "Bill", 15));
            list.Add(new Record(4, "Michael", 42));

            return list;
        }

    }
}

File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections
Imports System.Windows.Forms
Imports System.ComponentModel.Design
Imports DevExpress.XtraBars
Imports DevExpress.XtraReports.UserDesigner
' ...

Namespace FieldList_UpdateDataSource
	Partial Public Class Form1
		Inherits Form

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Const xmlPath As String = "..\..\Cars.xml"

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			xrDesignPanel1.ExecCommand(ReportCommand.NewReport)
		End Sub

		Private Sub commandBarItem46_ItemClick(ByVal sender As Object, ByVal e As ItemClickEventArgs) Handles barButtonItem1.ItemClick
			BindReportToData()
		End Sub

		#Region "#UpdateFieldList"
		Private Sub BindReportToData()
			If xrDesignPanel1.Report Is Nothing Then
				Return
			End If
			 ' Create a data source and bind it to a report.
			xrDesignPanel1.Report.DataSource = CreateDataSource()

			' Update the Field List.
			Dim fieldList As FieldListDockPanel = CType(xrDesignDockManager1(DesignDockPanelType.FieldList), FieldListDockPanel)
			Dim host As IDesignerHost = CType(xrDesignPanel1.GetService(GetType(IDesignerHost)), IDesignerHost)
			fieldList.UpdateDataSource(host)
		End Sub
		#End Region ' #UpdateFieldList

		Private Function CreateDataSource() As ArrayList
			' Create a list.
			Dim list As New ArrayList()

			' Populate the list with records.
			list.Add(New Record(1, "Jane", 19))
			list.Add(New Record(2, "Joe", 30))
			list.Add(New Record(3, "Bill", 15))
			list.Add(New Record(4, "Michael", 42))

			Return list
		End Function

	End Class
End Namespace